sunpy/map/tests/test_mapbase.py

```diff
@@ -1219,8 +1220,10 @@ def test_more_than_two_dimensions():
     hdr['TELESCOP'] = 'XXX'
     hdr['cunit1'] = 'arcsec'
     hdr['cunit2'] = 'arcsec'
-    with pytest.warns(SunpyUserWarning, match='This file contains more than 2 dimensions.'):
-        bad_map = sunpy.map.Map(bad_data, hdr)
+    with pytest.warns(SunpyMetadataWarning, match='Missing CTYPE1 from metadata, assuming CTYPE1 is HPLN-TAN'):
+        with pytest.warns(SunpyMetadataWarning, match='Missing CTYPE2 from metadata, assuming CTYPE2 is HPLT-TAN'):
+            with pytest.warns(SunpyUserWarning, match='This file contains more than 2 dimensions.'):
+                bad_map = sunpy.map.Map(bad_data, hdr)
```

Comment:
You can chain multiple context managers with a comma:

```
with A() as X, B() as Y, C() as Z:
    do_something()
```

probably nicer than *all* the indentation

Reply:
I made the first into one.